### PR TITLE
add viz for conversion nii2gif

### PR DESCRIPTION
**Pull request by arnaudbore**
Not 100% ready but it's a start

### CODE REVIEW

**Comment by surchs** [comment]
Thanks for the PR @arnaudbore. I think we can slim this down by a bit:
1. We can keep all the modules at the top level of the package. `nii2gif.py` from `viz` can just be moved there
2. `nii2gif.py` still has some CLI fluff we won't need because we don't intend to use it as a CLI. The only needed function there is `create_gif`
3. `load_and_prepare_image` and `create_mosaic` and `write_gif_normal` from `core.py` can live inside `nii2gif.py`
4. `config.py` is not needed as a separate module. These default values should be explicitly be part of the function definition (i.e. `def create_gif(pos_args, cmap='hot', slices=(1, 2, 3), ...)`
5. The only other place where these default values are used is in `create_mosaic` in `core.py`. This in turn is only called from `write_gif_normal` which passes all of these values through anyway. I think we should just let the default values from `create_mosaic` be passed through and not redefine them there as well.

So bottom line: I'd remove the CLI functionality (`argparse`), merge `core.py` and `nii2gif.py`, get rid of the `config.py` and make the default values explicit at the level of `create_gif`.

**Comment by surchs** [comment]
Ah, I just realized that I can add these as an actual review. Let me do that real quick.

File: dashQC_fmri/__about__.py [hunk 1]
```diff
@@ -29,7 +29,9 @@
     'matplotlib',
     'nilearn',
     'scipy',
-    'scikit-learn'
+    'scikit-learn',
+    'scikit-image',
```
Comment (surchs) [inline dashQC_fmri/__about__.py:33]: do you need `scikit-image`? I don't see an import anywhere.

File: dashQC_fmri/__about__.py [hunk 2]
```diff
@@ -29,7 +29,9 @@
...
+    'imageio'
```
Comment (surchs) [inline dashQC_fmri/__about__.py:34]: please also add to the Dockerfile

File: dashQC_fmri/viz/nii2gif.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+"""Main entry point."""
+
+import argparse
```
Comment (surchs) [inline dashQC_fmri/viz/nii2gif.py:3]: remove the CLI functionality since we won't use it

**Review by surchs — CHANGES_REQUESTED** [review]
- [x] `config.py` should be removed and the default values entered explicitly in the function definition
- [x] `core.py` should be merged with `nii2gif.py`
- [x] remove the CLI functionality from `nii2gif.py` as it's not needed

File: Dockerfile [hunk 4]
```diff
@@ -29,6 +29,7 @@ RUN conda install -y python=3.8.2 \
                      nilearn=0.6.2 \
                      matplotlib=3.1.3 \
                      joblib=0.15.1 \
+                     imageio=2.2.0 \
```
Comment (surchs) [inline Dockerfile:32]: this doesn't appear to be a compatible version with the other libraries. Can you please find a version that plays nice with the others.

File: dashQC_fmri/nii2gif.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    Returns
+    -------
+    new_img: numpy array
+
+    """
+
+    maxAxis = np.max(origShape[0:3])
+
+    if len(origShape) > 3:
+        if slices == DEFAULT_SLICES:
```
Comment (surchs) [inline dashQC_fmri/nii2gif.py:109]: this appears strange to me.
1. If you intend to ignore the default settings for 4D files, set the default to `None`. This is in general better than using a mutable object (`list`) as a default argument.
2. I assume this is meant to find the center of the brain. If the nifti is in MNI format, then the correct way to do that would be to use the inverse [affine matrix](https://nipy.org/nibabel/coordinate_systems.html#the-affine-matrix-as-a-transformation-between-spaces) to obtain the voxel coordinates of the desired MNI coordinates.

File: dashQC_fmri/nii2gif.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+    """
+    Create grayscale image.
+
+    Parameters
+    ----------
+    in_img: numpy array
+        Can be a 3D image or a 4D image.
+    origShape: list
+        Shape of the original image.
+    slices: list
```
Comment (surchs) [inline dashQC_fmri/nii2gif.py:95]: This appears to be left over from the original codebase and is no longer accurate. Could you update this?

File: dashQC_fmri/nii2gif.py [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+import os
+
+from imageio import mimwrite
+from matplotlib.cm import get_cmap
+import nibabel as nb
+import numpy as np
+import warnings  # mainly for ignoring imageio warnings
+warnings.filterwarnings("ignore")
+
+DEFAULT_SLICES = [40, 40, 40]
```
Comment (surchs) [inline dashQC_fmri/nii2gif.py:12]: see below. I think it'd be better to just go with `None` in the function definition for the default values and then check for that to set them later. They don't seem to be used for 4D files anyway.

File: dashQC_fmri/nii2gif.py (hunk 5 above)
Comment (arnaudbore) [reply dashQC_fmri/nii2gif.py:109]: I don't what I was thinking...yes either you provide the slices or it will use a default setting which is x/2 y/2 and z*2/3. Completely arbitrary behavior.

File: dashQC_fmri/nii2gif.py [hunk 8]
```diff
@@ -101,11 +101,10 @@ def create_mosaic(in_img, origShape, slices=None, slicesOrder='csa'):
     new_img: numpy array
 
     """
-
     maxAxis = np.max(origShape[0:3])
 
     if len(origShape) > 3:
-        if slices:
+        if not slices:
```
Comment (surchs) [inline dashQC_fmri/nii2gif.py:107]: This is not good behaviour for a `None` check. See [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations).
1. not explicit. `if slices is None` is better
2. not specific. a lot of empty objects are `False` when cast to boolean (see e.g. [here](https://amir.rachum.com/blog/2012/08/25/you-cant-handle-the-truth/)).

**Review by surchs — CHANGES_REQUESTED** [review]

**Review by surchs — COMMENTED** [review]
the rest looks good to me

**Review by surchs — COMMENTED** [review]
I made the `None` change. Ready to merge.
